### ISSUE TITLE
UPS carrier model getting error when create plugin in to Magento 2.3.3 compatibility

### ISSUE THREAD

**Issue by Bhavik-kumar** [magento/magento2#25390]
### Preconditions (*)
Magento 2.3.3 & 2.3-develop
with Developer mode.

### Steps to reproduce (*)
1. create plugin for `Magento\Ups\Model\Carrier` with custom di.xml
`app/code/<vender>/<module>/etc/di.xml`

```
<type name="Magento\Ups\Model\Carrier">
	<plugin name="fix-for-Ups" type="<vender>\<module>\Plugin\Carrier\Ups" sortOrder="1" />
</type>
```

2. create class as below mention
```
namespace <vender>\<module>\Plugin\Carrier;

class Ups
{
    public function afterCollectRates(\Magento\Ups\Model\Carrier $subject, $result, $request)
    {
       
        return $result;
    }
}

```
### Expected result (*)
Plugin should work properly with developer mode

### Actual result (*)
Getting below error
```
1 exception(s): Exception #0 (BadMethodCallException): Missing required argument $data of Magento\Ups\Model\Carrier.

Exception #0 (BadMethodCallException): Missing required argument $data of Magento\Ups\Model\Carrier.
```

As my understand issue is because of `Magento\Ups\Model\Carrier::__construct` last param should be null (newly added in to 2.3.3) **$proxyDeferredFactory** is not follow compatibility

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-22614` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hostep** [comment]
This will probably be fixed by https://github.com/magento/magento2/pull/25181

**Comment by VladimirZaets** [comment]
Hi @Bhavik-kumar. Thank you for your report. 
The issue has been fixed in magento/magento2#26130 by @Bartlomiejsz in 2.4-develop branch
Related commit(s):
 - [cfd5c22848f82f29f69cd42ffbcc8e09fa848568](https://github.com/magento/magento2/commit/cfd5c22848f82f29f69cd42ffbcc8e09fa848568)

The fix will be available with the upcoming 2.4.0 release.

**Comment by Bhavik-kumar** [comment]
is any one confirm that this is fixed with 2.3.4 ?

**Comment by ananth-iyer** [comment]
@Bhavik-kumar I have replicated the issue on 2.3.5-p2 and fix is to run the `setup:di:compile` again.
